### DISCUSSION TITLE
Running Custom Build and just serving a function

### DISCUSSION THREAD

**Discussion by KeKs0r** [vercel/vercel#4759] [Help]
I am trying to simple run a command `now build` and use the resulting javascript file as a handler to a function.

### 1. Approach: with functions
I tried building the function into the `/api` directory with this configuration.
```
  //package.json
   "now-build": "nx build graph-api-vercel --configuration=vercel"
  // vercel.json
  "functions": {
    "api/main.js": {
      "runtime": "@vercel/node@1.7.1",
    }
  },
  "routes": [{ "src": "/api/.*", "dest": "api/main.js" }]
```
Which results into this error: 
`The function for api/main.js can't be handled by any builder. Make sure it is inside the api/ directory.`
It seems that vercel checks for the existing of `api/main.js` before running the `now-build` script

### 2. Approach: with Builders
```
  //package.json
   "now-build": "nx build graph-api-vercel --configuration=vercel"
  // vercel.json
  "builds": [{ "src": "api/main.js", "use": "@vercel/node" }],
  "routes": [{ "src": "/api/.*", "dest": "api/main.js" }]
```
In this case the now-build command is not executed

### 3. Approach: just routes
```
  //package.json
   "now-build": "nx build graph-api-vercel --configuration=vercel"
  // vercel.json
  "routes": [{ "src": "/api/.*", "dest": "api/main.js" }]
```
--> seems to run the build command, but the serveless function is not defined. (routes is basically ignored)

I had a public folder with a dummy index.html for all those tests, and after removing it with just routes I am seing this error:
`No Output Directory named "public" found after the Build completed. You can configure the Output Directory in your project settings.`

How do I just run a custom build command and have the resulting handler as a serverless function?

**Comment by paulogdm** [comment]
Vercel should only host "Serverless Functions" as helpers. For that reason, I do need to recommend that advanced use-cases, like full-blown APIs, use another provider for those workloads.

Currently `@vercel/node` does not support a build step for the function itself. You can read more about the build step [here](https://vercel.com/docs/runtimes#advanced-usage/advanced-node-js-usage/custom-build-step-for-node-js).
If, however, you do need a specific case, you can look at [custom builders](https://vercel.com/docs/runtimes#advanced-usage/developing-your-own-runtime) and how to develop your own.
